OpenRA.Game/Graphics/WorldRenderer.cs

```diff
@@ -195,14 +195,14 @@ void GenerateOverlayRenderables()
 		// PERF: Avoid LINQ.
 		void GenerateAnnotationRenderables()
 		{
-			foreach (var a in World.ActorsWithTrait<IRenderAnnotations>())
+			World.ApplyToActorsWithTrait<IRenderAnnotations>((Actor actor, IRenderAnnotations trait) =>
```

Comment:
Its not necessary to specify all three types here. This could be either
```
World.ApplyToActorsWithTrait<IRenderAnnotations>((actor, trait) =>
```
or
```
World.ApplyToActorsWithTrait>((Actor actor, IRenderAnnotations trait) =>
```

IMO the first is easier to read.